src/Disruptor/BatchEventProcessor.cs

```diff
@@ -208,13 +207,17 @@ private void ProcessEvents()
                     // => The test is currently implemented on struct proxies. See BatchEventProcessor<T>.BatchStartAware and StructProxy.
                     // For some reason this also improves BatchEventProcessor performance for IBatchStartAware event handlers.
 
-                    _batchStartAware.OnBatchStart(availableSequence - nextSequence + 1);
+                    var diff = availableSequence - nextSequence;
 
-                    while (nextSequence <= availableSequence)
+                    if (diff >= 0)
```

Comment:
Shouldn't this line be after OnBatchStart?

Reply:
The comment above says that Java version has an if check to avoid the call if the batch is empty

Reply:
We could maybe remove the condition from `Disruptor.BatchEventProcessor.BatchStartAware.OnBatchStart` / `Disruptor.BatchEventProcessorFactory.DefaultBatchStartAware.OnBatchStart`.  What do you think?

Reply:
It's strange the condition was `!=0`, what to do with negative numbers there?

Reply:
I think it's just a style preference that was introduced in the .NET version the java version uses >=   https://github.com/LMAX-Exchange/disruptor/blob/master/src/main/java/com/lmax/disruptor/BatchEventProcessor.java#L160